framework/include/utils/MooseMeshElementConversionUtils.h

```diff
@@ -0,0 +1,201 @@
+
+/**
+ * Creates sets of four nodes indices that can form TET4 elements to replace the original PRISM6
+ * element.
+ * @param diagonal_direction A boolean value indicating the direction of the diagonal line of Face
+ * 2
+ * @param tet_face_indices A vector of vectors of the original face indices of the PRISM6 element
+ * corresponding to the faces of the newly created TET4 elements
+ * @return a vector of vectors of node indices that can form TET4 elements
+ */
+std::vector<std::vector<unsigned int>>
+tetNodesForPrism(const bool diagonal_direction,
+                 std::vector<std::vector<unsigned int>> & tet_face_indices);
+
+/**
+ * Rotate a PYRAMID5 element nodes to ensure that the node with the minimum id is the first node
+ * for the bottom face.
+ * @param min_id_index The index of the node with the minimum id for the bottom face
+ * @param face_rotation A vector to record the rotation of the faces of the PYRAMID5 element
+ * @return a vector of node indices that can form a PYRAMID5 element
+ */
+std::vector<unsigned int> nodeRotationPYRAMIND5(unsigned int min_id_index,
+                                                std::vector<unsigned int> & face_rotation);
+
+/**
+ * For a rotated nodes that can form a PYRAMID5 element, create a series of four-node set that can
+ * form TET4 elements to replace the original PYRAMID5 element. The QUAD4 face of the
+ * PYRAMID5 element will be split by the diagonal line that involves the node with the minimum id
+ * of that face.
+ * @param pyramid_nodes A vector of pointers to the nodes that can form a PYRAMID5 element
+ * @param rotated_tet_face_indices A vector of vectors of the original face indices of the
+ * PYRAMID5 element corresponding to the faces of the newly created TET4 elements
+ * @return a vector of vectors of pointers to the nodes that can form TET4 elements
+ */
+std::vector<std::vector<Node *>>
+pyramidNodesToTetNodesDeterminer(std::vector<Node *> & pyramid_nodes,
+                                 std::vector<std::vector<unsigned int>> & rotated_tet_face_indices);
+
+/**
+ * Convert all the elements in a 3D mesh consisting only linear elements into TET4 elements.
+ * @param mesh The mesh to be converted
+ * @param elems_to_process A vector of pairs of element ids and a bool indicating whether the
+ * element needs to be fully retained or will be partially cut in the following procedures
+ * @param converted_elems_ids_to_cut A vector of element ids that will be cut in the following
+ * procedures
+ * @param converted_elems_ids_to_retain A vector of element ids that will be fully retained in
+ * the following procedures
+ * @param block_id_to_remove The id of a new subdomain in the mesh containing all the elements to be
+ * removed
+ * @param delete_block_to_remove A bool indicating whether the block to be removed will be
+ * deleted in this method
+ */
+void convert3DMeshToAllTet4(ReplicatedMesh & mesh,
```

Comment:
Doing a partial conversion would cause issues as there are quad faces shared by two elements but only one of them is converted.
A potential solution would be to have a transition layer with tri faces on one side and quad faces on the other side. But the algorithm could be complex and non-trivial. I think we can figure it out in the future if needed.

Reply:
even if you do not do a partial conversion, getting rid of the cutting language can work for these arguments.

you just need to change from "will be cut" to "keep track of the id of these converted elements"